app/nexus/internal/initialization/recovery/recovery.go

```diff
@@ -163,6 +164,37 @@ func RecoverBackingStoreUsingKeeperShards(source *workloadapi.X509Source) {
 	}
 }
 
+// HydrateMemoryFromBackingStore loads all secrets from the persistent storage
+// into the application's memory state. This function is typically called during
+// application startup to restore the secret state from the previous session.
+//
+// The function reads all secrets from the backing store using
+// persist.ReadAllSecrets() and imports them into the application state using
+// state.ImportSecrets(). If no secrets are found in the backing store, the
+// function returns without making any changes to the application state.
+//
+// Example usage:
+//
+//	func initializeApp() {
+//		// Other initialization code...
+//		memory.HydrateMemoryFromBackingStore()
+//		// Continue with application startup
+//	}
+func HydrateMemoryFromBackingStore() {
+	const fName = "HydrateMemoryFromBackingStore"
+
+	log.Log().Info(fName, "msg", "HydrateMemoryFromBackingStore")
+
+	secrets := persist.ReadAllSecrets()
+	if len(secrets) == 0 {
+		return
+	}
+
+	state.ImportSecrets(secrets)
+
+	log.Log().Info(fName, "msg", "HydrateMemoryFromBackingStore: secrets loaded")
+}
```

Comment:
This is the majority of the magic.

`ImportSecrets()` is a new method introduced to the Go SDK.